konnectivity-client/go.mod

```diff
@@ -8,21 +8,21 @@ require (
 	github.com/prometheus/client_golang v1.11.1
 	go.uber.org/goleak v1.2.0
 	golang.org/x/net v0.17.0 // indirect
-	google.golang.org/grpc v1.40.0
```

Comment:
https://kubernetes.io/releases/#release-v1-26 *just* EOLd. So we're now supporting 1.27. We could bump go to 1.20. protobuf can move to 1.31.0 . genproto is hard to determine relative version. 1.27 uses v0.0.0-20230525234030-28d5490b6b19.

Reply:
>  We could bump go to 1.20. 

Done.

> protobuf can move to 1.31.0

It's currently at 1.30.0, but I'm reluctant for this branch to get ahead of all other branches. Could we start with this in `master`?